main.py

```diff
@@ -379,6 +380,53 @@ def is_better_title(new_title, old_title):
         )
 
 
+def top_locations(limit=20, kind=None):
+    """ Return a list of names and titles appearing recently in the news """
+
+    with SessionContext(commit=False) as session:
+        q = (
+            session.query(
+                Location.name,
+                Location.kind,
+                Location.country,
+                Location.article_url,
+                Location.latitude,
+                Location.longitude,
+                Article.id,
+                Article.heading,
+                Root.domain,
+            )
+            .join(Article)
+            .join(Root)
+            .filter(Root.visible)
+            .order_by(desc(Article.timestamp))
+        )
+
+    if kind:
+        q = q.filter(Location.kind == kind)
+
+    toplist = []
+    for l in q.limit(limit).all():
+        zoom = "6z" if l.kind == "country" else "16z"
+        mapurl = "http://google.com/maps/@{0},{1},{2}".format(l[4], l[5], zoom)
```

Comment:
Does `l.latitude, l.longitude` not work? And should it not be `https://` rather than `http://`?